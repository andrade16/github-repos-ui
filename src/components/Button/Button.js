import React from "react";
import "./Button.scss";

const Button = ({ label, color, tooltip, onClick }) => {
  return (
    <button
      className="button"
      style={{
        backgroundColor: color,
        borderColor: color,
        outlineColor: color,
      }}
      title={tooltip}
      onClick={onClick}
    >
      {label}
    </button>
  );
};

export default Button;
