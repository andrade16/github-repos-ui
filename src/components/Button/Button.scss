@import "src/utils/variables";

.button {
  cursor: $pointer-cursor;
  color: $white;
  border-radius: $small-border-radius;
  border-style: solid;
  height: 5vh;
  width: 10vw;
  min-width: 128px;
  min-height: 31px;
}
