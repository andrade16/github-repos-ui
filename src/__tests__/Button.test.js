import React from "react";
import renderer from "react-test-renderer";
import Button from "../components/Button/Button";

describe("<Button/>", () => {
  it("Renders <Button/> correctly", () => {
    const tree = renderer
      .create(
        <Button
          label="View Commits"
          color={"#007bff"}
          tooltip="tooltip Test"
          onClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
