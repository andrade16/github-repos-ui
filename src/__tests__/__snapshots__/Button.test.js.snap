// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Renders <Button/> correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#007bff",
      "borderColor": "#007bff",
      "outlineColor": "#007bff",
    }
  }
  title="tooltip Test"
>
  View Commits
</button>
`;
